### DISCUSSION TITLE
Objects memoization

### DISCUSSION THREAD

**Discussion by peinguin** [47ng/nuqs#521] [Q&A] [answered]
I made a custom parser, that parses and serializes a Date.
And this Date instance returns new even if a get param is not changed.
Is there any way, how to memoize an object in parser? Or to set some sort of condition, when the object is new?

**Comment by franky47** [comment, marked as answer]
We had this discussion recently with regards to re-renders when passing a non-primitive state to the dependency array of `useEffect` and the like:

https://github.com/47ng/nuqs/issues/516
